core/DataAccess/LogQueryBuilder/JoinGenerator.php

```diff
@@ -278,10 +283,16 @@ public function sortTablesForJoin($tA, $tB)
         }
 
         if (is_array($tA)) {
+            if (isset($tA['joinOn']) && is_string($tA['joinOn']) && strpos($tA['joinOn'] . '.', $tB) === 0) {
```

Comment:
The sorting was actually incorrect and did not check whether a joinOn is defined... in this case we need to change the sort order based on the join. Eg when  `on log_visit.idvisit = log_foobar.idvisit` then we need to make sure to sort  `log_foobar` before `log_visit` otherwise it cannot join `log_foobar` when it is only added later....